src/common/misc.c

```diff
@@ -1071,3 +1072,46 @@ daos_prop_copy(daos_prop_t *prop_req, daos_prop_t *prop_reply)
 	}
 	return rc;
 }
+
+static __thread struct {
+	uint64_t	host_id;
+	uint64_t	hlc;
+} dtx_id_tls;
+
+/* for the memcpy in daos_dti_gen */
+D_CASSERT(sizeof(((struct dtx_id *)0)->dti_uuid) >= sizeof(dtx_id_tls));
+
+static uint64_t
+daos_gethostid(void)
+{
+#ifndef HOST_NAME_MAX
+#define HOST_NAME_MAX 512
+#endif
+	char		hostname[HOST_NAME_MAX + 1] = { 0 };
+	int		rc;
+
+	rc = gethostname(hostname, HOST_NAME_MAX);
+	if (rc == 0)
+		return d_hash_murmur64((unsigned char *)hostname, HOST_NAME_MAX,
+				       609815U);
+
+	D_WARN("gethostname failed rc %d (errno %d).\n", rc, errno);
+	return (gethostid() << 32) | rand();
+}
+
+void
+daos_dti_gen(struct dtx_id *dti, bool zero)
+{
+	if (zero) {
+		memset(dti, 0, sizeof(*dti));
+	} else {
+		/* uuid_generate() is too slow to generate the uuid for dti_uuid
+		 * now use {host_id, hlc} to replace it.
+		 */
+		if (dtx_id_tls.hlc == 0)
+			dtx_id_tls.host_id = daos_gethostid();
```

Comment:
So different threads on the same client will get the same host_id, right? Why not keep the thread ID as you did in former patch?

Reply:
thread_id is not must needed, no? the original gethostid() returns same value on different boro nodes, so I changed to hash hostname.

Reply:
Our purpose is to make the DTX ID to be unique globally, in spite of which information is used. Here, if without the thread ID, then is it possible that the different threads on the same client may generate the same DTX ID for different RPCs by race? That depends on the HLC. It seems that the HLC will guarantee the atomic increasing under the race conditions. If yes, then it may be fine.

Reply:
right, I think the hlc is unique on same node (client or server), so seems enough for different nodes get different hostid to differentiate it.

Reply:
hmm, seems still need the thread_id somehow, I'll update it.

Reply:
HLC is per process only.

Reply:
> HLC is per process only.

right, I updated to add thread_id.
the daos_gethostid is still possible not unique, I feel still not safe enough. seems not easy to generate a unique hostid.

@Nasf-Fan another option is we still use uuid_generate() for dti_uuid, but each thread only calls it once so the overhead can be ignored. And each daos_dti_gen() will generate a new hlc for dti_sec.  i.e. the dtxs created by same thread will with same dti_uuid and different hlc, then needs to use both dti_uuid and dti_sec to identify the DTX, is it good?

Reply:
Yes, calling uuid_generate() once seems better solution.

Reply:
Currently, the dtx_id (that contains both the uuid_t and the time-stamp) is used as opaque value on server side except for the resent RPC check. So initializing the uuid_t per thread only once should work well.

Reply:
ok, thanks. I refreshed the patch just now.
Yesterday I wrote another version that uses mac address as hostid, but a problem is the IB/OPA mac address is with 20 bytes.
Now it is much simpler.